Develop3

Hemull184 opened the pull request:
Добавил локальный отчет Allure

Hemull184 committed:
add pom.xml

Hemull184 committed:
add HomePage

Hemull184 committed:
add PersonalAccountPage

Hemull184 committed:
add RegistrationPage

Hemull184 committed:
add README.md

Hemull184 committed:
add Yandex

Hemull184 committed:
изменил путь к драйверам

Hemull184 committed:
изменил путь к драйверам

Hemull184 committed:
поменял путь

Hemull184 committed:
убрал exe

Hemull184 committed:
вернул путь  с  \  и exe

Hemull184 committed:
добавил для linux

Hemull184 committed:
добавил для linux

Hemull184 committed:
очередная попытка с драйверами

Hemull184 committed:
добавил отчет allure

Hemull184 committed:
добавил отчет allure

nikolay-977 commented on src/main/java/pages/PersonalAccountPage.java:
```diff
@@ -0,0 +1,54 @@
+    }
+    @Step("Подожать появления кнопки 'Конструктор'")
+    public void waitConstructorButton() {
+        WebElement until = new WebDriverWait(driver, 5).until(ExpectedConditions.visibilityOfElementLocated(CONSTRUCTOR_BUTTON));
```
Можно улучшить: следует избегать повторяющегося кода, повторяющиеся строки лучше вынести в отдельный метод и в параметрах передавать таймаут и локатор

nikolay-977 commented on src/main/java/pages/RegistrationPage.java:
```diff
@@ -0,0 +1,53 @@
+    }
+    @Step("Запольнить поля регистрации")
+    public void registration(String name, String email, String password) {
+        filledName(name);
```
Отлично: применяется объединение методов.

nikolay-977 commented on src/main/java/user/RandomUser.java:
```diff
@@ -0,0 +1,18 @@
+        return new UserCreds(RandomStringUtils.randomAlphabetic(10) + "@yandex.ru", "123456789", RandomStringUtils.randomAlphabetic(10));
+    }
+
+    public static final String KEY_DRIVER = "webdriver.chrome.driver";
```
Необходимо исправит: не следует создавать класс для разных целей (генерация  кредов и хранения настроек для веб-драйвера). Вынеси настройки для веб-драйвера в одтельный класс WebDriverConfig и отдельный пакет config

nikolay-977 commented on src/test/java/HomePageTest.java:
```diff
@@ -0,0 +1,65 @@
+
+    HomePage objHomePage;
+
+    @Parameterized.Parameters
```
Необходимо исправить: тесты должны запускаться в двух браузерах: Google Chrome и Яндекс.Браузер. Можно реализовать: через параметр (файл .properties, переменная окружения, системная проперти, комманд лайн флаг), либо паттерн фабрика. **Недопустимо**: два комплекта тестов, различающихся браузером, закомментированный код, **браузер как параметр теста при параметризованных тестах.**

nikolay-977 commented on src/test/java/PersonalAccountTest.java:
```diff
@@ -0,0 +1,77 @@
+    }
+
+    @After
+    public void clean() {
```
Можно улучшить: вынести методы с аннотациями `@Before, @BeforeAll` и `@After, @AfterAll` в родительский класс. Это позволит избежать повторяющегося кода и сократит код автотестов.

Hemull184 committed:
1) Для расстановки пробелов и табуляции использовал сочетание клавиш Ctrl+Alt+L
2) Дополнил README.MD (как запустить тесты в яндекс браузере)
3) Повторяющиеся строки вынест в отдельный метод
4) Вынес настройки для веб-драйвера в одтельный класс
5) Исправил запуск тестов для разных браузеров (теперь тесты в яндекс браузер запускаются по команде mvn clean test -Dbrowser_type=yandex, а в хроме mvn clean test)
